src/Microsoft.Identity.Client/Cache/Keys/MsalAppMetadataCacheKey.cs

```diff
@@ -0,0 +1,68 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Identity.Client.Cache.Keys
+{
+    /// <summary>
+    /// App metadata is an optional entity in cache and can be used by apps to store additional metadata applicable to a particular client.
+    /// </summary>
+    internal class MsalAppMetadataCacheKey : IiOSKey
```

Comment:
>MsalAppMetadataCacheKey [](start = 19, length = 23)

Is this in sync with what iOS native does for the storage?

Reply:
Yes, Olga documented this pretty well. Also, the feature flag for FOCI for iOS and Android is set to false, meaning that: 

- we won't serialize app metadata
- we won't serialize the FamilyId field in the RT
- we won't make cache calls to check AppMetadat or FRT

Let me know if this makes sense - I tried to optimize for performance, i.e. eliminate cache calls. 
I can easily enable FOCI on iOS and Android (just a few simple methods to implement). On iOS, it should just work because the Cache Schema is documented. On Android not so much, but we can just choose a location for appmetadata and be done with it.